### PR TITLE
Added ansible >= 2.3.0 to documentation/enforcing

### PR DESCRIPTION
**Pull request by Fobhep**
This PR is meant to solve #56

### CODE REVIEW

File: modules/foreman_compute_profile.py [hunk 1]
```diff
@@ -47,9 +47,12 @@
     default: present
     choices: ["present", "absent"]
 notes:
-- Requires nailgun
+  - Requires nailgun
```
Comment (mdellweg) [inline modules/foreman_compute_profile.py]: This seems superfluous now.

File: modules/foreman_compute_profile.py [hunk 2]
```diff
@@ -65,6 +68,12 @@
 
 RETURN = ''' # '''
 
+from ansible.release import __version__ as ansible_version
+if [int(i) for i in ansible_version.split(".")] >= [2, 3, 0]:
+    HAS_PROPER_VERSION = True
+else:
+    HAS_PROPER_VERSION = False
+
```
Comment (mdellweg) [inline modules/foreman_compute_profile.py]: There is a logical error in this: When used with ansible < 2.3, the subsequent import of foreman_helper will still fail with an exception before we can report the incompatibility.

File: modules/foreman_compute_profile.py [hunk 3]
```diff
@@ -65,6 +68,12 @@
 
 RETURN = ''' # '''
 
+from ansible.release import __version__ as ansible_version
+if [int(i) for i in ansible_version.split(".")] >= [2, 3, 0]:
+    HAS_PROPER_VERSION = True
+else:
+    HAS_PROPER_VERSION = False
```
Comment (mdellweg) [inline modules/foreman_compute_profile.py]: You can simply assign the result of the comparison to the variable here.

**Comment by ekohl** [comment]
Rather than all the duplication, can't we introduce a shared helper that has this logic?

**Comment by mdellweg** [comment]
That is exactly the Problem: The module_utils thing is only available since Ansible 2.3. Before that there is no sharing.

**Comment by ekohl** [comment]
Pure python can import packages too. Would a relative import work?

**Comment by mdellweg** [comment]
These files must be known to ansible and copied to the Machine running the task. That is why we chose this module_utils directory in the first place.

**Comment by ekohl** [comment]
Ah yes, I keep forgetting that.

**Comment by Fobhep** [comment]
Yeah. It is really a shame we have to do this so not-DRY. But there is not really an alternative. @mdellweg  and me did a lot of digging before we wrote cement and could not find anything except the module_utils import

File: modules/foreman_compute_profile.py (hunk 2 above)
Comment (Fobhep) [reply modules/foreman_compute_profile.py]: yes, thanks :) I did not finish the PR yet, hence WIP ;)

**Comment by evgeni** [comment]
I'd prefer not to have this check in the modules, but only document the fact in `README.md` and `requirements.txt`.

**Comment by Fobhep** [comment]
I picked this PR up again and did not only add the Ansible >= 2.3 dependency but I also tried to sort of "normalize" the documentation of all modules. 
Meaning I fixed the identation and tried to unify certain things. Feedback is welcome! :)
@evgeni @akofink @mdellweg

File: modules/foreman_compute_profile.py [hunk 4]
```diff
@@ -22,7 +22,13 @@
 module: foreman_compute_profile
 short_description: Manage Foreman Compute Profiles using Foreman API
 description:
-- Create and delete Foreman Compute Profiles using Foreman API
+  - Create and delete Foreman Compute Profiles using Foreman API
+version_added: "2.0"
```
Comment (mdellweg) [inline modules/foreman_compute_profile.py]: I think, we should remove that until it is upstream.

File: modules/foreman_installation_medium.py [hunk 5]
```diff
@@ -27,7 +27,8 @@
 author:
   - "Manuel Bonk(@manuelbonk) ATIX AG"
 requirements:
-  - nailgun >= 0.16.0
+  - "nailgun >= 0.16.0"
```
Comment (akofink) [inline modules/foreman_installation_medium.py]: I'd be awesome if we could verify this, but I can't really think of a good way to do so.

File: modules/foreman_compute_profile.py (hunk 4 above)
Comment (Fobhep) [reply modules/foreman_compute_profile.py]: if everybody is ok, I would open an issue to  be solved before going actually upstream, that includes fixing the number to the actual upstream version.

File: modules/foreman_location.py [hunk 6]
```diff
@@ -22,45 +22,46 @@
...
+  - Manage Foreman Location
 author:
-    - "Matthias M Dellweg (@mdellweg) ATIX AG"
+  - "Matthias M Dellweg (@mdellweg) ATIX AG"
 requirements:
-    - "nailgun >= 0.28.0"
-    - "python >= 2.6"
+  - "nailgun >= 0.28.0"
+  - "python >= 2.6"
+  - ansible >= 2.3
```
Comment (evgeni) [inline modules/foreman_location.py]: should this be quoted too?

File: modules/foreman_installation_medium.py (hunk 5 above)
Comment (Fobhep) [reply modules/foreman_installation_medium.py]: yeah - me neither really.

File: modules/foreman_location.py (hunk 6 above)
Comment (Fobhep) [reply modules/foreman_location.py]: ah - yes it should. I will fix that

Comment (Fobhep) [reply modules/foreman_location.py]: @evgeni fixed

**Pull request merged**